### ISSUE TITLE
Nested helper renders $ASYNCID

### ISSUE THREAD

**Issue by johngully** [assemble/assemble#837]
I'm trying to get an assemble to generate a menu hierarchy.  My understanding is that the `pages` collection isn't available to the templates any longer an that the recommended approach is to create a custom helper.

I have created two helpers, and they each work independently.  However I need one to be nested within the other, and once I combine them the child intermittently returns an `{$ASYNCID$3$9$}`.  Neither of these helpers are doing any async work.

### Helpers
`assemble-helpers/menuCategories.js`
```js
module.exports = function (categories, options) {
  var value = '<ul>';
  // Logic to match pages to the specified categories list has been removed, and the issue is still present
  var page = { key: "page key", data: { title: "page title"}};
  value += '<li>' + options.fn(page) + '</li>';
  value += '</ul>';
  return value;
};
```

`assemble-helpers/menuItems.js`
```js
module.exports = function (category, options) {
  var value = '<ul>';
  // Logic to match pages to the specified category has been removed, and the issue is still present
  var item = { key: "item key", data: { title: "item title "}};
  value += '<li>' + options.fn(item) + '</li>';
  value += '</ul>';
  return value;
};
```

### Implementation
`template.hbs` 
```hbs
<nav>
{{#menuCategories this.patternStructure}}
  <a href="{{key}}">{{data.title}}</a>
  {{#menuItems .}}
    <a href="{{key}}">{{data.title}}</a>
  {{/menuItems}}
{{/menuCategories}}
</nav>
```

Assemble setup
_This has been modified to read more simply, so there's a possibility for a few typos_
```js
var options = {} // Removed for simplicity
var extname = require('gulp-extname');
var prettify = require('gulp-prettify');
var assemble = require('assemble');

var app = assemble();
app.helpers('/assemble-helpers/*.js');
app.layout('/layouts/test.hbs');
app.partials('/partials/*.hbs');
app.pages('/src/**/*.{hbs,md,tmpl}');

app.task('processTemplates', function(done) {
  processHandlebarsTemplates(options);
  processMarkdownTemplates(options)
  done();
}); 

app.build('processTemplates', done);

function processHandlebarsTemplates(options) {
  var ext = 'hbs';
  var engine = require('engine-handlebars');
  app.engine(ext, engine);
  app.src('/src/**/*.' + ext))
    .pipe(app.renderFile())
    .pipe(prettify())
    .pipe(extname())
    .pipe(app.dest('/dest'))
    .on('error',  console.log);
}

function processMarkdownTemplates(options) {
  var ext = 'md';
  var markdown = require('gulp-markdown');
  app.src('/src/**/*.' + ext))
    .pipe(markdown())
    .pipe(app.renderFile())
    .pipe(prettify())
    .pipe(app.dest('/dest'))
    .on('error', console.log);
}
```

### Generated output
Sometimes it works
```html
<nav class="library__nav">
  <ul>
    <li> <a href="page key">page title</a>
      <ul>
        <li> <a href="item key">item title</a> </li>
      </ul>
    </li>
  </ul>
</nav>
```
Othertimes it does not
```html
<nav class="library__nav">
  <ul>
    <li> <a href="page key">page title</a> {$ASYNCID$3$9$}</li>
  </ul>
</nav>
```

**Comment by assemblebot** [comment]
@johngully Thanks for the issue! If you're reporting a bug, please be sure to include:

- The version of `assemble` you are using.
- Your assemblefile.js (This can be in a gist)
- The commandline output. (Screenshot or gist is fine)
- What you expected to happen instead.

If your issue is related to one of the following, please open an issue there:
  - [grunt-assemble](https://github.com/assemble/grunt-assemble/issues/new?title=Nested%20helper%20renders%20%24ASYNCID&body=I'm%20trying%20to%20get%20an%20assemble%20to%20generate%20a%20menu%20hierarchy.%20%20My%20understanding%20is%20that%20the%20%60pages%60%20collection%20isn't%20available%20to%20the%20templates%20any%20longer%20an%20that%20the%20recommended%20approach%20is%20to%20create%20a%20custom%20helper.%0D%0A%0D%0AI%20have%20created%20two%20helpers%2C%20and%20they%20each%20work%20independently.%20%20However%20I%20need%20one%20to%20be%20nested%20within%20the%20other%2C%20and%20once%20I%20combine%20them%20the%20child%20intermittently%20returns%20an%20%60%7B%24ASYNCID%243%249%24%7D%60.%20%20Neither%20of%20these%20helpers%20are%20doing%20any%20async%20work.%0D%0A%0D%0A%23%23%23%20Helpers%0D%0A%60assemble-helpers%2FmenuCategories.js%60%0D%0A%60%60%60js%0D%0Amodule.exports%20%3D%20function%20(categories%2C%20options)%20%7B%0D%0A%20%20var%20value%20%3D%20'%3Cul%3E'%3B%0D%0A%20%20%2F%2F%20Logic%20to%20match%20pages%20to%20the%20specified%20categories%20list%20has%20been%20removed%2C%20and%20the%20issue%20is%20still%20present%0D%0A%20%20var%20page%20%3D%20%7B%20key%3A%20%22page%20key%22%2C%20data%3A%20%7B%20title%3A%20%22page%20title%22%7D%7D%3B%0D%0A%20%20value%20%2B%3D%20'%3Cli%3E'%20%2B%20options.fn(page)%20%2B%20'%3C%2Fli%3E'%3B%0D%0A%20%20value%20%2B%3D%20'%3C%2Ful%3E'%3B%0D%0A%20%20return%20value%3B%0D%0A%7D%3B%0D%0A%60%60%60%0D%0A%0D%0A%60assemble-helpers%2FmenuItems.js%60%0D%0A%60%60%60js%0D%0Amodule.exports%20%3D%20function%20(category%2C%20options)%20%7B%0D%0A%20%20var%20value%20%3D%20'%3Cul%3E'%3B%0D%0A%20%20%2F%2F%20Logic%20to%20match%20pages%20to%20the%20specified%20category%20has%20been%20removed%2C%20and%20the%20issue%20is%20still%20present%0D%0A%20%20var%20item%20%3D%20%7B%20key%3A%20%22item%20key%22%2C%20data%3A%20%7B%20title%3A%20%22item%20title%20%22%7D%7D%3B%0D%0A%20%20value%20%2B%3D%20'%3Cli%3E'%20%2B%20options.fn(item)%20%2B%20'%3C%2Fli%3E'%3B%0D%0A%20%20value%20%2B%3D%20'%3C%2Ful%3E'%3B%0D%0A%20%20return%20value%3B%0D%0A%7D%3B%0D%0A%60%60%60%0D%0A%0D%0A%23%23%23%20Implementation%0D%0A%60template.hbs%60%20%0D%0A%60%60%60hbs%0D%0A%3Cnav%3E%0D%0A%7B%7B%23menuCategories%20this.patternStructure%7D%7D%0D%0A%20%20%3Ca%20href%3D%22%7B%7Bkey%7D%7D%22%3E%7B%7Bdata.title%7D%7D%3C%2Fa%3E%0D%0A%20%20%7B%7B%23menuItems%20.%7D%7D%0D%0A%20%20%20%20%3Ca%20href%3D%22%7B%7Bkey%7D%7D%22%3E%7B%7Bdata.title%7D%7D%3C%2Fa%3E%0D%0A%20%20%7B%7B%2FmenuItems%7D%7D%0D%0A%7B%7B%2FmenuCategories%7D%7D%0D%0A%3C%2Fnav%3E%0D%0A%60%60%60%0D%0A%0D%0AAssemble%20setup%0D%0A_This%20has%20been%20modified%20to%20read%20more%20simply%2C%20so%20there's%20a%20possibility%20for%20a%20few%20typos_%0D%0A%60%60%60js%0D%0Avar%20extname%20%3D%20require('gulp-extname')%3B%0D%0Avar%20prettify%20%3D%20require('gulp-prettify')%3B%0D%0Avar%20assemble%20%3D%20require('assemble')%3B%0D%0A%0D%0Avar%20app%20%3D%20assemble()%3B%0D%0Aapp.helpers('%2Fassemble-helpers%2F*.js')%3B%0D%0Aapp.layout('%2Flayouts%2Ftest.hbs')%3B%0D%0Aapp.partials('%2Fpartials%2F*.hbs')%3B%0D%0Aapp.pages('%2Fsrc%2F**%2F*.%7Bhbs%2Cmd%2Ctmpl%7D')%3B%0D%0A%0D%0Aapp.task('processTemplates'%2C%20function(done)%20%7B%0D%0A%20%20var%20ext%20%3D%20'hbs'%3B%0D%0A%20%20var%20engine%20%3D%20require('engine-handlebars')%3B%0D%0A%20%20app.engine(ext%2C%20engine)%3B%0D%0A%20%20app.src('%2Fsrc%2F**%2F*.'%20%2B%20ext))%0D%0A%20%20%20%20.pipe(app.renderFile())%0D%0A%20%20%20%20.pipe(prettify())%0D%0A%20%20%20%20.pipe(extname())%0D%0A%20%20%20%20.pipe(app.dest('%2Fdest%2Fpattern-library'))%0D%0A%20%20%20%20.on('error'%2C%20%20console.log)%3B%0D%0A%20%20done()%3B%0D%0A%7D)%3B%20%0D%0A%0D%0Aapp.build('processTemplates'%2C%20done)%3B%0D%0A%60%60%60%0D%0A%0D%0A%23%23%23%20Generated%20output%0D%0ASometimes%20it%20works%0D%0A%60%60%60html%0D%0A%3Cnav%20class%3D%22library__nav%22%3E%0D%0A%20%20%3Cul%3E%0D%0A%20%20%20%20%3Cli%3E%20%3Ca%20href%3D%22page%20key%22%3Epage%20title%3C%2Fa%3E%0D%0A%20%20%20%20%20%20%3Cul%3E%0D%0A%20%20%20%20%20%20%20%20%3Cli%3E%20%3Ca%20href%3D%22item%20key%22%3Eitem%20title%3C%2Fa%3E%20%3C%2Fli%3E%0D%0A%20%20%20%20%20%20%3C%2Ful%3E%0D%0A%20%20%20%20%3C%2Fli%3E%0D%0A%20%20%3C%2Ful%3E%0D%0A%3C%2Fnav%3E%0D%0A%60%60%60%0D%0AOthertimes%20it%20does%20not%0D%0A%60%60%60html%0D%0A%3Cnav%20class%3D%22library__nav%22%3E%0D%0A%20%20%3Cul%3E%0D%0A%20%20%20%20%3Cli%3E%20%3Ca%20href%3D%22page%20key%22%3Epage%20title%3C%2Fa%3E%20%7B%24ASYNCID%243%249%24%7D%3C%2Fli%3E%0D%0A%20%20%3C%2Ful%3E%0D%0A%3C%2Fnav%3E%0D%0A%60%60%60%0D%0A) Issues with using assemble in grunt or the grunt-assemble library.
  - [handlebars-helpers](https://github.com/assemble/handlebars-helpers/issues/new?title=Nested%20helper%20renders%20%24ASYNCID&body=I'm%20trying%20to%20get%20an%20assemble%20to%20generate%20a%20menu%20hierarchy.%20%20My%20understanding%20is%20that%20the%20%60pages%60%20collection%20isn't%20available%20to%20the%20templates%20any%20longer%20an%20that%20the%20recommended%20approach%20is%20to%20create%20a%20custom%20helper.%0D%0A%0D%0AI%20have%20created%20two%20helpers%2C%20and%20they%20each%20work%20independently.%20%20However%20I%20need%20one%20to%20be%20nested%20within%20the%20other%2C%20and%20once%20I%20combine%20them%20the%20child%20intermittently%20returns%20an%20%60%7B%24ASYNCID%243%249%24%7D%60.%20%20Neither%20of%20these%20helpers%20are%20doing%20any%20async%20work.%0D%0A%0D%0A%23%23%23%20Helpers%0D%0A%60assemble-helpers%2FmenuCategories.js%60%0D%0A%60%60%60js%0D%0Amodule.exports%20%3D%20function%20(categories%2C%20options)%20%7B%0D%0A%20%20var%20value%20%3D%20'%3Cul%3E'%3B%0D%0A%20%20%2F%2F%20Logic%20to%20match%20pages%20to%20the%20specified%20categories%20list%20has%20been%20removed%2C%20and%20the%20issue%20is%20still%20present%0D%0A%20%20var%20page%20%3D%20%7B%20key%3A%20%22page%20key%22%2C%20data%3A%20%7B%20title%3A%20%22page%20title%22%7D%7D%3B%0D%0A%20%20value%20%2B%3D%20'%3Cli%3E'%20%2B%20options.fn(page)%20%2B%20'%3C%2Fli%3E'%3B%0D%0A%20%20value%20%2B%3D%20'%3C%2Ful%3E'%3B%0D%0A%20%20return%20value%3B%0D%0A%7D%3B%0D%0A%60%60%60%0D%0A%0D%0A%60assemble-helpers%2FmenuItems.js%60%0D%0A%60%60%60js%0D%0Amodule.exports%20%3D%20function%20(category%2C%20options)%20%7B%0D%0A%20%20var%20value%20%3D%20'%3Cul%3E'%3B%0D%0A%20%20%2F%2F%20Logic%20to%20match%20pages%20to%20the%20specified%20category%20has%20been%20removed%2C%20and%20the%20issue%20is%20still%20present%0D%0A%20%20var%20item%20%3D%20%7B%20key%3A%20%22item%20key%22%2C%20data%3A%20%7B%20title%3A%20%22item%20title%20%22%7D%7D%3B%0D%0A%20%20value%20%2B%3D%20'%3Cli%3E'%20%2B%20options.fn(item)%20%2B%20'%3C%2Fli%3E'%3B%0D%0A%20%20value%20%2B%3D%20'%3C%2Ful%3E'%3B%0D%0A%20%20return%20value%3B%0D%0A%7D%3B%0D%0A%60%60%60%0D%0A%0D%0A%23%23%23%20Implementation%0D%0A%60template.hbs%60%20%0D%0A%60%60%60hbs%0D%0A%3Cnav%3E%0D%0A%7B%7B%23menuCategories%20this.patternStructure%7D%7D%0D%0A%20%20%3Ca%20href%3D%22%7B%7Bkey%7D%7D%22%3E%7B%7Bdata.title%7D%7D%3C%2Fa%3E%0D%0A%20%20%7B%7B%23menuItems%20.%7D%7D%0D%0A%20%20%20%20%3Ca%20href%3D%22%7B%7Bkey%7D%7D%22%3E%7B%7Bdata.title%7D%7D%3C%2Fa%3E%0D%0A%20%20%7B%7B%2FmenuItems%7D%7D%0D%0A%7B%7B%2FmenuCategories%7D%7D%0D%0A%3C%2Fnav%3E%0D%0A%60%60%60%0D%0A%0D%0AAssemble%20setup%0D%0A_This%20has%20been%20modified%20to%20read%20more%20simply%2C%20so%20there's%20a%20possibility%20for%20a%20few%20typos_%0D%0A%60%60%60js%0D%0Avar%20extname%20%3D%20require('gulp-extname')%3B%0D%0Avar%20prettify%20%3D%20require('gulp-prettify')%3B%0D%0Avar%20assemble%20%3D%20require('assemble')%3B%0D%0A%0D%0Avar%20app%20%3D%20assemble()%3B%0D%0Aapp.helpers('%2Fassemble-helpers%2F*.js')%3B%0D%0Aapp.layout('%2Flayouts%2Ftest.hbs')%3B%0D%0Aapp.partials('%2Fpartials%2F*.hbs')%3B%0D%0Aapp.pages('%2Fsrc%2F**%2F*.%7Bhbs%2Cmd%2Ctmpl%7D')%3B%0D%0A%0D%0Aapp.task('processTemplates'%2C%20function(done)%20%7B%0D%0A%20%20var%20ext%20%3D%20'hbs'%3B%0D%0A%20%20var%20engine%20%3D%20require('engine-handlebars')%3B%0D%0A%20%20app.engine(ext%2C%20engine)%3B%0D%0A%20%20app.src('%2Fsrc%2F**%2F*.'%20%2B%20ext))%0D%0A%20%20%20%20.pipe(app.renderFile())%0D%0A%20%20%20%20.pipe(prettify())%0D%0A%20%20%20%20.pipe(extname())%0D%0A%20%20%20%20.pipe(app.dest('%2Fdest%2Fpattern-library'))%0D%0A%20%20%20%20.on('error'%2C%20%20console.log)%3B%0D%0A%20%20done()%3B%0D%0A%7D)%3B%20%0D%0A%0D%0Aapp.build('processTemplates'%2C%20done)%3B%0D%0A%60%60%60%0D%0A%0D%0A%23%23%23%20Generated%20output%0D%0ASometimes%20it%20works%0D%0A%60%60%60html%0D%0A%3Cnav%20class%3D%22library__nav%22%3E%0D%0A%20%20%3Cul%3E%0D%0A%20%20%20%20%3Cli%3E%20%3Ca%20href%3D%22page%20key%22%3Epage%20title%3C%2Fa%3E%0D%0A%20%20%20%20%20%20%3Cul%3E%0D%0A%20%20%20%20%20%20%20%20%3Cli%3E%20%3Ca%20href%3D%22item%20key%22%3Eitem%20title%3C%2Fa%3E%20%3C%2Fli%3E%0D%0A%20%20%20%20%20%20%3C%2Ful%3E%0D%0A%20%20%20%20%3C%2Fli%3E%0D%0A%20%20%3C%2Ful%3E%0D%0A%3C%2Fnav%3E%0D%0A%60%60%60%0D%0AOthertimes%20it%20does%20not%0D%0A%60%60%60html%0D%0A%3Cnav%20class%3D%22library__nav%22%3E%0D%0A%20%20%3Cul%3E%0D%0A%20%20%20%20%3Cli%3E%20%3Ca%20href%3D%22page%20key%22%3Epage%20title%3C%2Fa%3E%20%7B%24ASYNCID%243%249%24%7D%3C%2Fli%3E%0D%0A%20%20%3C%2Ful%3E%0D%0A%3C%2Fnav%3E%0D%0A%60%60%60%0D%0A) Issues with using handlebars helpers from the handlebars-helpers library.

**Comment by jonschlinkert** [comment]
@johngully thanks for the detailed issue. Do you know if the issue was happening before this weekend, or did you just discover the issue today?

**Comment by jonschlinkert** [comment]
also, if you don't mind trying something to help narrow down the issue, could you try doing `npm install templates@0.15.2` in your project, and see if that resolves it. thanks

**Comment by jonschlinkert** [comment]
also, I noticed that you're doing `app.pages()` but also using `app.src()` with the same pattern, so both are loading the same files. 

Instead of using `app.src()` try using `app.toStream('pages')`

**Comment by johngully** [comment]
@jonschlinkert I just got this issue today.  I'll try the step you suggested (moving the `app.layouts`, etc inside a task, changing to `app.toStream('pages')` and updating the teamplates version.  I'll respond back with my findings.

Thanks for the super fast response!

**Comment by jonschlinkert** [comment]
no problem, thanks for letting us know!

**Comment by jonschlinkert** [comment]
also are you using watch by any chance?

**Comment by johngully** [comment]
I am not using watch

**Comment by johngully** [comment]
### More information
@jonschlinkert I found some more information that I believe will lead to the resolution of this issue.  _I updated the original post to better reflect reality.  Please review the original code._

I'm actually making multiple passes at template processing for different file types.  I chose to  implement it this way to support support both handlebars and markdown templates (I'm open to another approach if possible).

If I comment out either the markdown or handlebars functions then the templates processed by that function will render correctly.  Processing them together seems to introduce the problem.

### Updated code still fails
Based upon this new information and your suggested approach I did the following:
- Removed the use of `app.pages()` (not currently needed, but may need to revisit).
- Created a new task with all the "setup" steps
- Moved each template processor to it's own task

_Each of these steps were tested independently, with no success_

### Revised code example
```js
...

app.task('setup', function(done) {
  app.helpers('/assemble-helpers/*.js');
  app.layout('/layouts/test.hbs');
  app.partials('/partials/*.hbs');
  app.pages('/src/**/*.{hbs,md,tmpl}');
});

app.task('processTemplates', ['processHandlebarsTemplates', 'processMarkdownTemplates'], function(done) {
  // processHandlebarsTemplates(options);
  // processMarkdownTemplates(options); 
  done();
});

app.task('processHandlebarsTemplates', ['setup'], function(done) {
  processHandlebarsTemplates(options);
  done();
});

app.task('processMarkdownTemplates', ['setup'], function(done) {
  processMarkdownTemplates(options);
  done();
});

...
```

**Comment by jonschlinkert** [comment]
the two process tasks might run in parallel, which could cause rendering issues. you will need to ensure that they run in series, by using something like async. but I'm not sure if it's related to your issue.

**Comment by johngully** [comment]
I would actually prefer just to have a single task if possible.  Is there a way to configure a single run that will process both markdown and handlebars files?  Last time I looked there was not an "engine" for markdown...

**Comment by johngully** [comment]
I also updated the templates dependency without any success
```bash
npm install templates@0.15.2 --save
```

**Comment by johngully** [comment]
The last thing on the list was to use `app.toStream('pages')`.  I'm struggling a little with the syntax.  Can you provide some insight.  The following code does not work:

```js
function processHandlebarsTemplates(options) {
  var ext = 'hbs';
  var engine = require('engine-handlebars');
  app.engine(ext, engine);
  app.pages('/src/**/*.' + ext));
  app.toStream('pages')
    .pipe(app.renderFile())
    .pipe(prettify())
    .pipe(extname())
    .pipe(app.dest('/dest'))
    .on('error',  console.log);
}

// This change has also been made for the markdown function
```

**Comment by doowb** [comment]
@johngully you can [helper-markdown](https://github.com/helpers/helper-markdown#assemble) in handlebars to render the markdown files in the same pass as the handlebars files.

I'm working on an example of how I would setup the `assemblefile`

**Comment by doowb** [comment]
Example of rendering all the pages in one pass:

```js

// register the handlebars engine for all extensions
app.engine(['hbs', 'md', 'tmpl'], require('engine-handlebars'));

// register markdown helper for markdown files
app.helper('markdown', require('helper-markdown'));

// set the layout on markdown files so they're converted to html
app.onLoad(/\.md/, function(file, next) {
  file.layout = 'markdown';
  next();
});

app.task('setup', function(done) {
  app.helpers('/assemble-helpers/*.js');
  app.layouts('/layouts/*.hbs');
  app.partials('/partials/*.hbs');
  app.pages('/src/**/*.{hbs,md,tmpl}');
});

app.task('processTemplates', ['setup'], function() {
  return app.toStream('pages')
    .pipe(app.renderFile())
    .pipe(prettify())
    .pipe(extname())
    .pipe(app.dest('/dest'))
    .on('error',  console.log);
});
```

You'll need to add a `markdown.hbs` template to your layouts folder that looks like this:

```handlebars
{{#markdown}}
{% body %}
{{/markdown}}
```

The main change that I think was causing issues, is that I'm returning the stream in the task. This lets the task know when the stream ends so it can continue on. In your examples above, you were calling `done` before the stream had a chance to finish, which could have caused rendering issues.

The other change was adding an `onLoad` middleware for `.md` files. This way you can add the `markdown` layout to only the markdown files, but still render with the handlebars engine.

**Comment by jonschlinkert** [comment]
> I'm returning the stream in the task

completely slipped past me, good catch @doowb

**Comment by johngully** [comment]
I just made a simple change to get a single pass working and the helper issues went away.  I added the `helper-markdown` and modified the handlebars processor function to the following:

```js
app.helper('markdown', require('helper-markdown'));

function processHandlebarsTemplates(options) {
  var ext = 'hbs';
  var engine = require('engine-handlebars');
  app.engine(ext, engine);
  app.pages('/src/**/*.{hbs,md}'));
  app.toStream('pages')
    .pipe(app.renderFile())
    .pipe(prettify())
    .pipe(extname())
    .pipe(app.dest('/dest'))
    .on('error',  console.log);
}
```

**Comment by jonschlinkert** [comment]
@johngully you will still need to return the stream as @doowb did in his example, otherwise the task does not know when to finish

**Comment by johngully** [comment]
Good catch.  I didn't notice that.  I'll do that as well.

I guess my ultimate question now is, do we think there is a problem when multiple tasks are run concurrently.  Or was I just doing it wrong?

**Comment by jonschlinkert** [comment]
> do we think there is a problem when multiple tasks are run concurrently. Or was I just doing it wrong?

I guess a little of both. technically you were doing it wrong, so there is nothing "wrong" with the application itself.  But given that you're not the first experienced programmer to have the issue, and I didn't even catch the cause (despite the detailed code example and great overview you gave) that tells me we also have "implementation problem". 

The same issues occur with gulp. hopefully we'll come up with a way to make this more transparent to users.

**Comment by johngully** [comment]
@doowb I think I can get things rolling from here.  Thanks again for the support!  Do you guys have a Slack channel for the project (if not you should definitely consider getting one, it's super helpful for fielding these conversations)?

I guess my last request would be for and example that illustrates how I could have multiple tasks.  One for processing handlebars and another for processing markdown (similar to my original approach).  I think that piece of the puzzle is still missing for me.  Perhaps an example will clarify.

**Comment by jonschlinkert** [comment]
closing since I think this is resolved. 

@johngully we do have a [gitter channel](https://gitter.im/assemble/assemble), we use Slack a lot though. I'd love to get a slack channel started.

> One for processing handlebars and another for processing markdown (similar to my original approach).

can you explain a bit more about what you'd like to do and expected results? (eg. that isn't covered by @doowb's example)
